src/builder.ts

```diff
@@ -516,11 +515,16 @@ export class NexusPrismaBuilder {
     const dedupScalarNames = [...new Set(allScalarNames)]
     const scalars: any[] = []
 
-    if (dedupScalarNames.includes('DateTime')) {
+    // FIXME The type of .type above is nexus.core.AllOutputTypes
+    // but this does not account for custom scalars. Nexus
+    // should change its AllOutputTypes type, or export a new type,
+    // that integrates custom scalers. Conversely, nexus-prisma
+    // typegen should contribute (e.g. via interface merging) to it
+    // the scalars found in DMMF.
+    if (dedupScalarNames.includes('DateTime' as any)) {
       scalars.push(dateTimeScalar)
     }
-
-    if (dedupScalarNames.includes('UUID')) {
+    if (dedupScalarNames.includes('UUID' as any)) {
       scalars.push(uuidScalar)
     }
 
```

Comment:
Actual bug finds!

Reply:
> the type system won't be able to catch them at "dev time" anyway as there's no nexus generated typegen file yet.

Do you mean dev time of nexus-prisma?

> In the end, IMO, what matters here is that the surface/public API remain strongly typed.

agree

> Internally though, we cannot properly type everything as some parts rely on the generated typegen file which happens at runtime.

My conclusion was different. We can import types from `nexus` and correctly type our plugin.

Reply:
@Weakky let's pick up the topic of thread next week.